src/webgpu/api/validation/createComputePipeline.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { ValidationTest } from './validation_test.js';
+
+type TShaderStage = 'compute' | 'vertex' | 'fragment' | 'empty';
+
+class F extends ValidationTest {
+  getShaderModule(
+    shaderStage: TShaderStage = 'compute',
+    entryPoint: string = 'main'
+  ): GPUShaderModule {
+    let code;
+    switch (shaderStage) {
+      case 'compute': {
+        code = `[[stage(compute)]] fn ${entryPoint}() {}`;
+        break;
+      }
+      case 'vertex': {
+        code = `
+        [[stage(vertex)]] fn ${entryPoint}() -> [[builtin(position)]] vec4<f32> {
+          return vec4<f32>(0.0, 0.0, 0.0, 1.0);
+        }`;
+        break;
+      }
+      case 'fragment': {
+        const fragColorType = 'i32';
+        const suffix = '';
+        code = `
+        [[stage(fragment)]] fn ${entryPoint}() -> [[location(0)]] vec4<${fragColorType}> {
+          return vec4<${fragColorType}>(0${suffix}, 1${suffix}, 0${suffix}, 1${suffix});
+        }`;
+        break;
+      }
+      case 'empty':
+      default: {
+        code = '';
+        break;
+      }
+    }
+    return this.device.createShaderModule({ code });
+  }
+
+  getInvalidShaderModule(): GPUShaderModule {
+    this.device.pushErrorScope('validation');
+    const code = 'deadbeaf'; // Something make nonsense
+    const shaderModule = this.device.createShaderModule({ code });
+    this.device.popErrorScope();
+    return shaderModule;
+  }
+
+  doCreateComputePipelineTest(
+    isAsync: boolean,
+    _success: boolean,
+    descriptor: GPUComputePipelineDescriptor
+  ) {
+    if (isAsync) {
+      if (_success) {
+        this.shouldResolve(this.device.createComputePipelineAsync(descriptor));
+      } else {
+        this.shouldReject('OperationError', this.device.createComputePipelineAsync(descriptor));
+      }
+    } else {
+      if (_success) {
+        this.device.createComputePipeline(descriptor);
+      } else {
+        this.expectValidationError(() => {
+          this.device.createComputePipeline(descriptor);
+        });
+      }
+    }
+  }
+}
+
+export const g = makeTestGroup(F);
+
+g.test('basic_use_of_createComputePipeline')
+  .cases(poptions('isAsync', [true, false]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+    t.doCreateComputePipelineTest(isAsync, true, {
+      compute: { module: t.getShaderModule('compute', 'main'), entryPoint: 'main' },
+    });
+  });
+
+g.test('shader_module_must_be_valid')
+  .cases(poptions('isAsync', [true, false]))
+  .fn(async t => {
+    const { isAsync } = t.params;
+    t.doCreateComputePipelineTest(isAsync, false, {
+      compute: {
+        module: t.getInvalidShaderModule(),
+        entryPoint: 'main',
+      },
+    });
+  });
+
+g.test('shader_module_stage_must_be_compute')
+  .cases(
+    params()
+      .combine(poptions('isAsync', [true, false]))
+      .combine(
+        poptions('shaderModuleStage', [
+          'compute' as TShaderStage,
```

Comment:
I think adding `as const` after the list might be sufficient instead of all the `as TShaderStage`. Though I don't understand why ^^

Reply:
I change it into "TShaderStage[]".